### DISCUSSION TITLE
How does bucketing actually work? If I turn it off will it cropped my image?

### DISCUSSION THREAD

**Discussion by askladd-anon** [bmaltais/kohya_ss#2861] [Q&A]
I don't actually understand how bucketing work. All my images is formatted with sdxl training resolution native size, which is
-  1024 x 1024
-  1152 x 896
-  896 x 1152
-  1216 x 832
-  832 x 1216
-  1344 x 768
-  768 x 1344
-  1536 x 640
-  640 x 1536

Yet, when I activate bucketing it will still put my image into buckets that is not in these resolutions for some reason. So I decided to turn off bucketing, since from this sentence below from the documentation, I assume it will not do anything to my image, if for each aspect ratio, all the images have the same reso.

```
Enlargement and reduction are performed while maintaining the aspect ratio of the image. If the aspect ratio is not the same as the standard size, the vertical or horizontal size of the image after scaling may exceed the standard size. For example, if the base size is 512x512 ( 1 aspect ratio ) and the image size is 1536x1024 ( 1.5 aspect ratio ), the image will be scaled down to 768x512 ( 1.5 aspect ratio remains).
```

But I notice, that if i use batch size more than 1, the resulting lora will output the subject out of frame for some reason. I didn't notice it when using batch size=1. So, If i don't turn on bucketing, will kohya crop all my images to 1:1, or not? How do i make sure kohya do not touch my images at all? How does batching work, if some aspect ratio only have one images for example, will it crop another images to fit to its aspect ratio, if i turn off bucketing?

**Comment by daileta1** [comment]
If you don't turn on buckets, everything gets cropped to the same size and 1:1 ratio.  

I combed through the bucketing code and came up with these guidelines for how to make sure your image will get cropped correctly to make the best LoRA: https://civitai.com/articles/7777/detailed-flux-training-guide-dataset-preparation#cropping-and-aspect-ratios-skae1og5m

That same guide should also answer your questions about batching and how it works with bucketing.

**Reply by Tekaiguy** [reply]
Thank you so much for the guide, do you know if you can train SDXL on 4k images or will that cause too many adverse effects?
